tests/test_regression.py

```diff
@@ -0,0 +1,124 @@
+"""
+    Suite of tests that go through the entire process of creating a new fktable
+    from a empty folder.
+
+    The target theory is 400 and the relevant `.toml`, theory runcard and eko template
+    are downloaded from https://github.com/NNPDF/theories during this test so this tests
+    has the double effect of ensuring compatibility between both repositories.
+"""
+from pathlib import Path
+from subprocess import run
+from urllib.request import urlretrieve
+
+import numpy as np
+import pytest
+from pineappl.fk_table import FkTable
+from yaml import dump, safe_load
+
+THEORIES_REPO = "https://raw.githubusercontent.com/NNPDF/theories/main"
+THEORYID = 400
+
+
+def _download_resources(filename, tmp_path):
+    """Download resources (filename) from the theories repo and put it in the same path
+    relative to THEORIES_REPO
+    """
+    output_file = tmp_path / filename
+    output_file.parent.mkdir(exist_ok=True)
+    urlretrieve(f"{THEORIES_REPO}/{filename}", output_file)
+    return output_file
+
+
+def _download_dataset(dataset, theoryid, tmp_path):
+    """Download both the yaml file and all grids for a given dataset for a given theory"""
+    yaml_file = f"data/yamldb/{theoryid}/{dataset}.yaml"
+    _download_resources(yaml_file, tmp_path)
+    # However, the yaml file goes into the "root" of the yaml directory, move it there!
+    right_yaml = tmp_path / "data" / "yamldb" / f"{dataset}.yaml"
+    (tmp_path / yaml_file).rename(right_yaml)
+    # Download the relevant grids for this dataset
+    res = safe_load(right_yaml.open("r", encoding="utf-8"))
+    grids = [i for j in res["operands"] for i in j]
+    for grid_name in grids:
+        _download_resources(f"data/grids/400/{grid_name}.pineappl.lz4", tmp_path)
+    return grids
+
+
+class _FakePDF:
+    """A Fake lhapdf-like PDF to evolve the grids"""
+
+    def __init__(self):
+        pids = np.arange(-6, 7, dtype=int)
```

Comment:
```suggestion
        pids = np.arange(-6, 7)
```
Nothing wrong, but maybe you're just too used to tf and Numba :)